### DISCUSSION TITLE
My thoughts after using Tempo as my primary music app for a week

### DISCUSSION THREAD

**Discussion by SteveDinn** [CappielloAntonio/tempo#116] [Ideas]
I just discovered Tempo about a week ago and ive been using it as my primary music app ever since. I've been looking for an alternative to the aging DSub and i think Tempo is the best contender so far. It's great!

I can also totally appreciate where you're coming from as a solo developer on an ambitious project like this one.

I'd like to put forward my list of features that I would want in my ultimate music player app. Take them all with a grain of salt; I realize these are all from my personal wishlist and may not fit into your vision of the project. I know this is a lot to drop on you, but I wanted to get my ideas down and see what others thought about them.

Feel free to close this issue and open whatever individual feature issues you'd like 

## Be able to hide and re-order the sections on the main page
As far as I can tell, the main screen has the following sections on the main page:
- Discovery
- Made for you
- Starred tracks
- Starred albums
- New releases
- Flashback
- Most played
- Last played
- Recently added

Personally, I would like to see *Recently added* first, followed by *Last played* as those are the albums I would most likely be looking to play when I open the app. I think that the order of these sections, as well as whether they are shown at all, should be definable by the user.

## Consistent context menu items
This was one of my main gripes with DSub: the items from the context menu were not consistent.

### Menu items for a track anywhere a track is seen
These might already be consistent in this app...
- Play Next
- Add to queue (play last)
- Rate
- Download
- Add to playlist
- Go to album (if not viewing the album already)
- Go to artist (if not viewing the artist already)

### Menu items for any collection of tracks
This category would include an album, a group of albums, an artist, a playlist, etc., anywhere where a list of tracks can be obtained.
- Play now
- Play next
- Add to queue (play last)
- Shuffle now (shuffles the tracks and then "Play now")
- Shuffle next (shuffles the tracks and then "Play next")
- Shuffle to queue (shuffles the tracks and then "Add to queue")
- Download (individually download each track)
- Add to playlist

### Albums-centric view
From the main page, I can see several options on albums' context menus that are not on the upper-right "..." menu when in the album-centric view. i.e., there is no way to "Play next" or "Add to queue" from the album-centric view. I think these menus should be the same as on the main page.

### Artist-centric view
- You could think about an artist as a chronological collection of all the tracks in all their albums, so the artist is ultimately a collection of tracks and should have the menu items that any collection of tracks does.
- Collapse the biography by default. Being able to read about the band is nice, but 
- IMO, the albums of an artist should be the right under the collapsed biography. They should be the first thing the user sees when viewing the artist. For me, "Most played tracks" is interesting, but it's not the most important thing.

### Horizontal scrolling
Maybe an unpopular opinion, but I don't think horizontal scrolling works well on a phone. It might be better on a tablet. I would rather see an abbriviated list, so it doesnt take up too much vertical space, and have to click through to a separate view to see the full list in something like a 2-column grid that I can scroll vertically though.

**Comment by jeffgt14** [comment]
Agree completely with all of these. Most are minor inconveniences though that can hopefully be refined over time. Android Auto is still the big one holding me back from using consistently but this app has the most promise of all the Android clients currently.

**Comment by dryya** [comment]
The big one I'm missing is go to album. I would love to see that functionality added.

**Comment by thom-github** [comment]
I'd like to add my voice here. Please make rating songs easier.

So far, all I've found is how to rate a song in a playlist. And even that's a bit of a pain (in the song list tap exactly the three dots next to the song, pick rate, rate song) - and I haven't found any other way to do it. Not in the song window, not when playing a whole album.

In short, every song should be rateable anywhere in the the UI. :)
